src/wasm.h

```diff
@@ -71,6 +71,7 @@ struct Name : public cashew::IString {
   Name() : cashew::IString() {}
   Name(const char *str) : cashew::IString(str) {}
   Name(cashew::IString str) : cashew::IString(str) {}
+  Name(const std::string &str) : cashew::IString(str.c_str()) {}
```

Comment:
need to pass `, false` to that IString constructor. The second param is "can I reuse this memory", and it's optimized for the initial use case of IStrings, which was to intern strings as we go, starting from a large existing buffer of text.

Or, perhaps we should flip that default. It means we can't merge code to emscripten's optimizer, but maybe that's not too bad at this point.